multiversx_sdk/core/transactions_outcome_parsers/delegation_transactions_outcome_parser.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+
+from multiversx_sdk.core.address import Address
+from multiversx_sdk.core.errors import ParseTransactionOutcomeError
+from multiversx_sdk.core.transactions_outcome_parsers.delegation_transactions_outcome_parser_types import \
+    CreateNewDelegationContractOutcome
+from multiversx_sdk.core.transactions_outcome_parsers.resources import (
+    TransactionEvent, TransactionOutcome, find_events_by_identifier)
+from multiversx_sdk.network_providers.constants import DEFAULT_ADDRESS_HRP
+
+
+class DelegationTransactionsOutcomeParser:
+    def __init__(self) -> None:
+        pass
+
+    def parse_create_new_delegation_contract(
+            self,
+            transaction_outcome: TransactionOutcome
+    ) -> List[CreateNewDelegationContractOutcome]:
+        self._ensure_no_error(transaction_outcome.logs.events)
+
+        events = find_events_by_identifier(transaction_outcome, "SCDeploy")
+        return [CreateNewDelegationContractOutcome(self._extract_contract_address(event)) for event in events]
+
+    def _ensure_no_error(self, transaction_events: List[TransactionEvent]) -> None:
```

Comment:
Is duplicated now, we should think of a refactoring (for a future PR).

Reply:
I was thinking the same thing. Will do in the next PR.